docs/src/tutorials/manifold.md

```diff
@@ -0,0 +1,198 @@
+# [How to Implement your own manifold](@id manifold-tutorial)
+
+```@meta
+CurrentModule = ManifoldsBase
+DocTestSetup  = quote
+    using Manifolds
+end
+```
+
+This tutorial demonstrates, how to easily set up your own manifold within `Manifolds.jl`.
+
+## Introduction
+
+If you looked around a little and saw the [interface](../interface.md), the amount of functions and possibilities, it might seem that a manifold might take some time to implement.
+This tutorial demonstrated, that you can get your first own manifold quite fast and you only have to implement the functions you actually need. For this tutorial it would be helpful, if you take a look at our [notation](../notation.md).
+For this tutorial it is helpful, if you heard of the exponential map, tangent vectors and the dimension of a manifold.
+
+We start with two technical preliminaries. If you want to start directly, you can [skip](@ref manifold-tutorial-task) this paragraph and revisit it for two of the implementation details.
+
+After that, we will
+
+* [model](@ref manifold-tutorial-task) the manifold
+* [implement](@ref manifold-tutorial-checks) two tests, so that points and tangent vectors can be checked for validity, for example also within [`ValidationManifold`](@ref),
+* [implement](@ref manifold-tutorial-fn) two functions, the exponential map and the manifold dimension.
+
+## [Technical preliminaries](@id manifold-tutorial-prel)
+
+There is only two small  technical things we require.
+First of all our [`Manifold`](@ref)`{𝔽}` has a parameter `𝔽`.
+This parameter indicates the [`number_system`](@ref) the manifold is based on, e.g. real-valued (`ℝ`) or complex-valued (`ℂ`) data.
+This can be used to simultaneously implement both a real-valued and a complex-valued type of a manifold.
```

Comment:
If you have `Circle{R}` and `Circle{C}` that's easier to do a common implementation for some functions compared to introducing a common abstract type.

That might not have been, how we _got_ to that parametrization, but it's a good reason anyways.

Reply:
Well, you can have a number system parameter in manifold types without having it as a parameter of `Manifold`. So a number system parameter of, let's say, `Symmetric` is there to avoid adding another abstract type while a number system parameter of the `Manifold` type it derives from is there to have `number_of_coordinates` for free.

`Circle` may not be the best example here because it overwrites `number_of_coordinates`.

Reply:
Having it as a parameter was a choice, yes. Still I think the `number_of_coordinates` is not be the main reasons for our manifolds (it was to pull it up to Manifold, yes).

In general our manifolds have this parameter for easy dispatch per field type. So what do you propose to write in the tutorial?

Reply:
In a basic tutorial like this I think it would be enough to say that for real manifolds users need to put `ManifoldsBase.ℝ` there and not worry about it, and for complex manifolds tell the reader that it has important interactions with bases.

Reply:
So I could just leave out line 31? That would be fine with me.

Reply:
I have added a suggested change below.

Reply:
Thanks for the careful review.